test/integration/targets/nxos_ip_interface/tests/cli/sanity.yaml

```diff
@@ -0,0 +1,79 @@
+---
+- debug: msg="START TRANSPORT:CLI nxos_ip_interface sanity test"
+
+- set_fact: testint1="{{ nxos_int1 }}"
+- set_fact: testint2="{{ nxos_int2 }}"
+
+- name: "Put interface {{testint1}} into default state"
+  nxos_config: &intdefault1
+    lines:
+      - "default interface {{ testint1 }}"
+    provider: "{{ cli }}"
+  ignore_errors: yes
+
+- name: "Put interface {{testint2}} into default state"
+  nxos_config: &intdefault2
+    lines:
+      - "default interface {{ testint2 }}"
+    provider: "{{ cli }}"
+  ignore_errors: yes
+
+- name: "Make {{testint1}} a layer3 interface"
+  nxos_interface: &l3int1
+    interface: "{{ testint1 }}"
+    mode: layer3
+    description: 'Configured by Ansible - Layer3'
+    admin_state: 'up'
+    state: present
+    provider: "{{ cli }}"
+
+- name: "Make {{testint2}} a layer3 interface"
+  nxos_interface: &l3int2
+    interface: "{{ testint2 }}"
+    mode: layer3
+    description: 'Configured by Ansible - Layer3'
+    admin_state: 'up'
+    state: present
+    provider: "{{ cli }}"
+
+- name: Ensure ipv4 address is configured
+  nxos_ip_interface: &ipv4
+    interface: "{{ testint1 }}"
+    version: v4
+    state: present
+    addr: 20.20.20.20
+    mask: 24
+    provider: "{{ cli }}"
+  register: result
+
+- assert: &true
+    that:
+      - "result.changed == true"
```

Comment:
@Qalthos ... I would strongly discourage the use of titanium for further continued testing. It is not a platform we will support. Frankly, there are a lot of issues with the ansible issues right now and I'd rather put my time and attention to fixing those issues. Having said that, you are more than welcome to calibrate the tests for titanium.

Regarding the improvements you are looking for:-
- both interfaces are put into the default state prior to bring up. This clears any state on the interface including the ip address. This works on all hardware platforms and n9k virtual platforms.
- I'm not sure I understand. That scenario would be taken care of just by idempotence right ?
In general these are basic IT we have added to just get bare minimal coverage. We have not reached the point where we are doing in depth testing. At some point in the future the tests will be enhanced.